e2e/ibc_to_cctp_test.go

```diff
@@ -21,146 +21,326 @@
+	)
+}
 
-	// Generate orbiter payload
-	destinationDomain := uint32(0)
-	mintRecipient := testutil.RandomBytes(32)
-	destinationCaller := testutil.RandomBytes(32)
-	passthroughPayload := []byte("")
+func testIbcFailingUnsupportedAction(
+	t *testing.T,
+	ctx context.Context,
+	cdc codec.Codec,
+	s *Suite,
+	dstUsdcDenom string,
+	toOrbiterChannelID string,
+) {
+	amountToSend := math.NewInt(OneE6)
 
-	forwarding, err := forwarding.NewCCTPForwarding(
-		destinationDomain,
-		mintRecipient,
-		destinationCaller,
-		passthroughPayload,
+	forwarding, err := forwardingtypes.NewCCTPForwarding(
+		uint32(0),
+		testutil.RandomBytes(32),
+		testutil.RandomBytes(32),
+		[]byte(""),
 	)
 	require.NoError(t, err)
 
+	// Create a wrapped payload for the IBC memo without the required forwarding info.
 	feeRecipientAddr := testutil.NewNobleAddress()
-	feeAttr := action.FeeAttributes{
-		FeesInfo: []*action.FeeInfo{
-			{
-				Recipient:   feeRecipientAddr,
-				BasisPoints: 100,
-			},
+	action, err := actiontypes.NewFeeAction(
+		&actiontypes.FeeInfo{
+			Recipient:   feeRecipientAddr,
+			BasisPoints: 100,
 		},
-	}
+	)
+	action.Id = core.ACTION_SWAP
+	require.NoError(t, err)
+
+	p := core.PayloadWrapper{Orbiter: &core.Payload{
+		PreActions: []*core.Action{action},
+		Forwarding: forwarding,
+	}}
+	payloadBz, err := types.MarshalJSON(cdc, &p)
+	require.NoError(t, err)
 
-	action := core.Action{
-		Id: core.ACTION_FEE,
+	height, err := s.IBC.CounterpartyChain.Height(ctx)
+	require.NoError(t, err)
+
+	transfer := ibc.WalletAmount{
+		Address: OrbiterModuleAddr,
+		Denom:   dstUsdcDenom,
+		Amount:  amountToSend,
 	}
-	err = action.SetAttributes(&feeAttr)
+	ibcTx, err := s.IBC.CounterpartyChain.SendIBCTransfer(
+		ctx,
+		toOrbiterChannelID,
+		s.IBC.CounterpartySender.KeyName(),
+		transfer,
+		ibc.TransferOptions{
+			Memo: string(payloadBz),
+		},
+	)
+	require.NoError(t, err)
+	s.FlushRelayer(t, ctx, toOrbiterChannelID)
+
+	msg, err := interchainutil.PollForAck(
+		ctx,
+		s.IBC.CounterpartyChain,
+		height,
+		height+10,
+		ibcTx.Packet,
+	)
 	require.NoError(t, err)
 
-	payload, err := core.NewPayloadWrapper(forwarding, []*core.Action{&action})
+	expectedAck := &channeltypes.Acknowledgement{}
+	err = cdc.UnmarshalJSON(msg.Acknowledgement, expectedAck)
 	require.NoError(t, err)
+	require.Contains(
+		t,
+		expectedAck.GetError(),
+		"orbiter-middleware error",
+		"expected the error in the ack to contains the orbiter middleware error",
+	)
 
-	encCfg := testutil.MakeTestEncodingConfig("noble")
-	orbiter.RegisterInterfaces(encCfg.InterfaceRegistry)
-	payloadStr, err := types.MarshalJSON(encCfg.Codec, payload)
+	resp, err := s.IBC.CounterpartyChain.GetBalance(
+		ctx,
+		s.IBC.CounterpartySender.FormattedAddress(),
+		dstUsdcDenom,
+	)
 	require.NoError(t, err)
+	require.Equal(
+		t,
+		math.NewInt(OneE6),
+		resp,
+		"expected the address on the counterparty chain to have funds unlocked",
+	)
```

Comment:
would be more explicit to mention that it's the sent amount here

```suggestion
	require.Equal(
		t,
		amountToSend,
		resp,
		"expected the address on the counterparty chain to have funds unlocked",
	)
```